File: libs/trading_models/enums.py
"""TypeScript-style enums and literals for trading system."""

from enum import Enum
from typing import Literal


class Timeframe(str, Enum):
    """Trading timeframes."""
    M15 = "15m"
    H1 = "1h"
    H4 = "4h"
    D1 = "1d"


class TradingAction(str, Enum):
    """Trading actions."""
    BUY = "buy"
    SELL = "sell"
    HOLD = "hold"
    CLOSE_LONG = "close_long"
    CLOSE_SHORT = "close_short"


class Direction(str, Enum):
    """Trade direction."""
    LONG = "long"
    SHORT = "short"


class OrderType(str, Enum):
    """Order types."""
    MARKET = "market"
    LIMIT = "limit"
    STOP = "stop"
    STOP_LIMIT = "stop_limit"


class OrderStatus(str, Enum):
    """Order status."""
    PENDING = "pending"
    OPEN = "open"
    FILLED = "filled"
    PARTIALLY_FILLED = "partially_filled"
    CANCELLED = "cancelled"
    REJECTED = "rejected"
    EXPIRED = "expired"


class PatternType(str, Enum):
    """Pattern types."""
    SUPPORT_RESISTANCE = "support_resistance"
    BREAKOUT = "breakout"
    TREND_REVERSAL = "trend_reversal"
    PIN_BAR = "pin_bar"
    ENGULFING = "engulfing"
    DOJI = "doji"
    DIVERGENCE = "divergence"


class MarketRegime(str, Enum):
    """Market regime types."""
    BULL = "bull"
    BEAR = "bear"
    SIDEWAYS = "sideways"


# TypeScript-style literals
TimeframeLiteral = Literal["15m", "1h", "4h", "1d"]
TradingActionLiteral = Literal["buy", "sell", "hold", "close_long", "close_short"]
DirectionLiteral = Literal["long", "short"]
OrderTypeLiteral = Literal["market", "limit", "stop", "stop_limit"]
OrderStatusLiteral = Literal["pending", "open", "filled", "partially_filled", "cancelled", "rejected", "expired"]

File: libs/rust-common/src/trading_models/enums.rs
//! Trading enums compatible with Python models.

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum Timeframe {
    #[serde(rename = "15m")]
    M15,
    #[serde(rename = "1h")]
    H1,
    #[serde(rename = "4h")]
    H4,
    #[serde(rename = "1d")]
    D1,
}

impl Timeframe {
    pub fn as_str(&self) -> &'static str {
        match self {
            Self::M15 => "15m",
            Self::H1 => "1h",
            Self::H4 => "4h",
            Self::D1 => "1d",
        }
    }
    
    pub fn from_str(s: &str) -> Option<Self> {
        match s {
            "15m" => Some(Self::M15),
            "1h" => Some(Self::H1),
            "4h" => Some(Self::H4),
            "1d" => Some(Self::D1),
            _ => None,
        }
    }
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum TradingAction {
    Buy,
    Sell,
    Hold,
    CloseLong,
    CloseShort,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum Direction {
    Long,
    Short,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum OrderType {
    Market,
    Limit,
    Stop,
    StopLimit,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum OrderStatus {
    Pending,
    Open,
    Filled,
    PartiallyFilled,
    Cancelled,
    Rejected,
    Expired,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum PatternType {
    SupportResistance,
    Breakout,
    TrendReversal,
    PinBar,
    Engulfing,
    Doji,
    Divergence,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum MarketRegime {
    Bull,
    Bear,
    Sideways,
}